x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/action/TransportPutRollupJobAction.java

```diff
@@ -115,6 +125,26 @@ public void onFailure(Exception e) {
         });
     }
 
+    /**
+     * Ensure that the index pattern doesn't match any pre-existing rollup indices.  We don't need to check
+     * if the pattern would match it's own rollup index because the request already validated that.
+     */
```

Comment:
Maybe we can be less strict and only forbid mixing rollup and non-rollup indices in the same pattern ?
It should be fine to create the job if the pattern matches rollup indices only. Although I wonder if the problem of mixing indices (rollup and non-rollup) is not caught by the field validation. As you mentioned they have different field names so the validation of field names should fail all the time ?

Reply:
Good point about the field name validation, although i'm still a bit worried about unintended consequences of matching against a rollup index.  Barring the "rollup a rollup" usecase, it doesn't seem wise to let users match against an existing rollup if we can help it?

But maybe it's not worth the extra complexity?  If it matches another rollup index, that index will exist so we can validate against the fields and it should throw an exception.  It also doesn't protect against future rollup indices from matching the defined pattern, so we'd be relying on field level validation there anyway.

I think I'm convinced.  Will make the changes, should simplify the PR a bunch :)

Reply:
So, it turns out to be a bit more complicated.  The field-level validation is broken, due to how field caps behaves.  

Field caps only tells you the caps for the fields that match, in the resolved indices.  But it doesn't tell you about the indices that were resolved but _didn't_ match any field.  So we won't know the index pattern matched some random index without the correct fields.

That's unrelated to this change to this PR, but needs to be fixed for us to simplify.  Going to see what it takes, we may have to manually resolve the indices anyway to check.  Will keep you updated.

Reply:
Good catch. We can also add the matching indices in field caps's response. Could be per field : the list of index that doesn't have a definition for the field or global: the list of indices that match the pattern and then you can infer which index is missing from the field's response. I think it's a valid feature to add even if we don't use it in rollups.